django/urls/resolvers.py

```diff
@@ -346,11 +476,13 @@ def _is_callback(self, name):
         return name in self._callback_strs
 
     def resolve(self, path):
+        if not self._populated:
+            self._populate()  # This checks for some common errors
```

Comment:
@sjoerdjob, do you remember about why you added these lines? I don't spot any test failures with them removed. I thought maybe they were relevant to something else we removed along the way.

Reply:
Sorry. Currently I don't recall and don't have the ability to check. Probably indeed relevant to something that has since been removed/reverted.

Reply:
I added it [in this commit](https://github.com/knbk/django/commit/94b9bebb56ebab322ea5677f969d0763f785c295#diff-b2f90a810a553411112f412de2a26617R498). I do recall some test failures without that line, but it might not be necessary anymore.

Reply:
This was the test failure:
```
======================================================================
ERROR: test_contains_tuple_not_url_instance (urlpatterns.tests.InvalidURLsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.5/unittest/case.py", line 600, in run
    testMethod()
  File "/home/vagrant/django/tests/urlpatterns/tests.py", line 77, in test_contains_tuple_not_url_instance
    resolve('/')
  File "/home/vagrant/django/django/urls/base.py", line 27, in resolve
    return get_resolver(urlconf).resolve(path)
  File "/home/vagrant/django/django/urls/resolvers.py", line 505, in resolve
    sub_match = pattern.resolve(new_path)
AttributeError: 'tuple' object has no attribute 'resolve'

----------------------------------------------------------------------
```

This test has now been removed, I guess because the check for `BaseURL` instances has been removed from `_populate()`. I think it's safe to remove.